File: DS -py/fibbonacci_recursion.py
#Fibbonacci series: 0,1,1,2,3,5,...
def fibb(n):
    if n<=1:
        return n;
    dp[n]=fibb(n-1)+fibb(n-2)
    return dp[n];

dp=[0 for x in range(0,1000)]
# or can use dp=[0]*1000
dp[1]=1
print(fibb(6))

File: DS/fibbonacci_recursion.cpp
//Fibbonaci series: 0,1,1,2,3,5,8......

#include<bits/stdc++.h>
using namespace std;
int dp[1000];
int fibb(int n){
 if(n<=1) return n;
 dp[n]=fibb(n-1)+fibb(n-2);
 return dp[n];

}
int main(){
int n=5;
memset(dp,0,sizeof(dp));
// or can use dp[1000]={0}
dp[1]=1;
cout<<fibb(n-1)<<endl;

cout<<"Generating series: \n";
for(int i=0;i<n;i++)
cout<<fibb(i)<<"\n";
}